scikit-learn doesn't build on 21.11

### Describe the bug

I try to install scikit-learn on nixos 21.11 but `nixos-rebuild switch --upgrade` fails when building the package. Of the test of scikit-learn does fails and the installation fails:

```console
=================================== FAILURES ===================================
__________________________ test_rank_deficient_design __________________________
[gw0] linux -- Python 3.9.6 /nix/store/wl02plhc6zf84m6x9984l42wnnnbly5m-python3-3.9.6/bin/python3.9

    def test_rank_deficient_design():
        # consistency test that checks that LARS Lasso is handling rank
        # deficient input data (with n_features < rank) in the same way
        # as coordinate descent Lasso
        y = [5, 0, 5]
        for X in (
                  [[5, 0],
                   [0, 5],
                   [10, 10]],
                  [[10, 10, 0],
                   [1e-32, 0, 0],
                   [0, 0, 1]]
                 ):
            # To be able to use the coefs to compute the objective function,
            # we need to turn off normalization
            lars = linear_model.LassoLars(.1, normalize=False)
            coef_lars_ = lars.fit(X, y).coef_
            obj_lars = (1. / (2. * 3.)
                        * linalg.norm(y - np.dot(X, coef_lars_)) ** 2
                        + .1 * linalg.norm(coef_lars_, 1))
            coord_descent = linear_model.Lasso(.1, tol=1e-6, normalize=False)
            coef_cd_ = coord_descent.fit(X, y).coef_
            obj_cd = ((1. / (2. * 3.)) * linalg.norm(y - np.dot(X, coef_cd_)) ** 2
                      + .1 * linalg.norm(coef_cd_, 1))
>           assert obj_lars < obj_cd * (1. + 1e-8)
E           assert 5.233333333333334 < (0.5711500859559147 * (1.0 + 1e-08))

/nix/store/58kb7bq69grq085hac2gf92c0r1nnqyf-python3.9-scikit-learn-0.24.1/lib/python3.9/site-packages/sklearn/linear_model/tests/test_least_angle.py:248: AssertionError
```



### Steps To Reproduce
1. Enable scikit-learn for python3 in `confguration.nix`
2. `nixos-rebuild switch --upgrade`

### Expected behavior

scikit-learn should install without errors.

### Screenshots

n/a

### Additional context

It looks that hydra fails to build this package too:

```console
% hydra-check --channel=21.11 python39Packages.scikit-learn 
Build Status for nixpkgs.python39Packages.scikit-learn.x86_64-linux on 21.11
✖ (Failed) python3.9-scikit-learn-0.24.1 from 2021-12-24 - https://hydra.nixos.org/build/162040197

Last Builds:
✔ python3.9-scikit-learn-0.24.1 from 2021-12-12 - https://hydra.nixos.org/build/161393984
✔ python3.9-scikit-learn-0.24.1 from 2021-11-22 - https://hydra.nixos.org/build/159132659
```


### Notify maintainers

@DavHau 

### Metadata

```console
% nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.84, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.334934.8a053bc2255"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/scikit-learn/default.nix
+++ b/pkgs/development/python-modules/scikit-learn/default.nix
@@ -34,6 +34,12 @@ buildPythonPackage rec {
       url = "https://github.com/scikit-learn/scikit-learn/commit/e7ef22c3ba2334cb3b476e95d7c083cf6b48ce56.patch";
       sha256 = "174554k1pbf92bj7wgq0xjj16bkib32ailyhwavdxaknh4bd9nmv";
     })
+    # This patch fixes numerical issue with openblas 0.3.18. It was merged in scikit-learn v1.0.2.
+    # See details in https://github.com/scikit-learn/scikit-learn/issues/21340
+    ( fetchpatch {
+      url = "https://github.com/scikit-learn/scikit-learn/commit/657454c0e5a73071ec6681a0774fdaa4269adda7.patch";
+      sha256 = "1lv71ashr9wcgxqvqazrdpnm1l6cazcsj2sf6n9ybadpghrqp8yf";
+    })
   ];
 
   buildInputs = [

